csrc/ir/interface_nodes.h

```diff
@@ -527,6 +527,11 @@ class TensorView : public Val {
     return promote_reuse_;
   }
 
+  // Returns whether a TensorView has its first non-reduction axis parallelized
+  // on Didx
+  // Checks that the other non-reduction axis are not parallelized on Didx
+  bool isSharded() const;
```

Comment:
nit: I'd prefer this to be put in some util header for multidevice to make the definition of `TensorView` concise.

Reply:
done! I created the files `multidevice/utils.(h|cpp)` which will be populated with other functions as we progress